rust/noosphere-cli/tests/helpers/random.rs

```diff
@@ -0,0 +1,35 @@
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+/// A helper to support consistent use of seeded randomness in tests.
+/// Its primary purpose is to retain and report the seed in use for a
+/// random number generator that can be shared across threads in tests.
+/// This is probably not suitable for use outside of tests.
+pub struct TestEntropy {
+    seed: [u8; 32],
+    rng: Arc<Mutex<StdRng>>,
+}
+
+impl Default for TestEntropy {
+    fn default() -> Self {
+        Self::from_seed(rand::thread_rng().gen::<[u8; 32]>())
+    }
+}
+
+impl TestEntropy {
+    pub fn from_seed(seed: [u8; 32]) -> Self {
+        tracing::info!(?seed, "Initializing test entropy...");
```

Comment:
I've not seen that ? Prefix before. Is it for optional params.

Reply:
It's a `tracing`-specific syntax for logging just a value that implements `Debug`. It just appears as a field called `seed` in the log line.